store/videoTrack.ts

```diff
@@ -0,0 +1,96 @@
+import { atom } from "jotai";
+import {
+  atomWithReset,
+  useResetAtom,
+  useAtomValue,
+  useUpdateAtom,
+} from "jotai/utils";
+import type {
+  VideoTrackProps,
+  VideoTrackSchema,
+} from "$server/models/videoTrack";
+
+const toVideoTrackSchema = (
+  { language, content }: VideoTrackProps,
+  index: number
+): VideoTrackSchema => ({
+  id: index,
+  kind: "subtitles",
+  language,
+  url: URL.createObjectURL(new Blob([content])),
+});
+
+const isVideoTrackProps = (
+  videoTrack: VideoTrackProps | VideoTrackSchema
+): videoTrack is VideoTrackProps => "content" in videoTrack;
+
+const videoTracksPropsAtom = atomWithReset<VideoTrackProps[]>([]);
+const videoTracksPropsToSchemaAtom = atom<VideoTrackSchema[]>((get) =>
+  get(videoTracksPropsAtom).map(toVideoTrackSchema)
+);
+const videoTracksSchemaAtom = atom<VideoTrackSchema[]>([]);
+
+const videoTracksAtom = atom<VideoTrackSchema[]>((get) => {
+  if (get(videoTracksPropsAtom).length > 0)
+    return get(videoTracksPropsToSchemaAtom);
+  else return get(videoTracksSchemaAtom);
```

Comment:
videoTracksPropsToSchemaAtom はここでしか使われておらずあえてatomにするのは過剰では無いでしょうか?

```suggestion
  const videoTracksProps = get(videoTracksPropsAtom);
  return videoTracksProps.length > 0
    ? videoTracksProps.map(toVideoTrackSchema)
    : get(videoTracksSchemaAtom);
```

まだ全体を追えておらず間違った指摘かもですが…。

Reply:
> videoTracksPropsToSchemaAtom はここでしか使われておらずあえてatomにするのは過剰では無いでしょうか?

その通りだと思いました！修正しました